orchestration/templates/process-and-reingest-release.yaml

```diff
@@ -72,12 +72,16 @@ spec:
                   value: {{ $gcsPrefix | quote }}
             {{- $stagingDataset := "{{tasks.ingest-archive.outputs.parameters.dataset-name}}" }}
 
-          # If we have provenance information, cut a snapshot of the new data
+          # If we have provenance information and a snapshot doesn't exist, cut a snapshot of the new data
           # for the release-date.
           {{- if $versionIsPinned }}
-          - name: publish-processing-results
+          - name: check-latest-snapshot
             dependencies: [ingest-archive]
-            when: {{ $processingNeeded | quote }}
+            template: check-latest-snapshot
+          - name: publish-processing-results
```

Comment:
Getting my head around this a bit: so we're guarding against the situation where the first attempt at a snapshot succeeded but the rest of this ingest job failed for whatever reason and had no processing_history row created?

Reply:
Not quite; let's say a release was put out on Monday. The pipeline runs, processes, and then fails during the snapshot creation. Tuesday, it sees that processing isn't needed, but a snapshot doesn't exist, so we need to make a snapshot.